hal/ia32/cpu.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * CPU related routines
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <syspage.h>
+
+
+/* FIXME: temporary usage until new syspage implementation */
+extern int hal_memoryGetEntry(unsigned int *offs, mem_entry_t *entry);
+
+
+__attribute__((noreturn)) void hal_cpuJump(addr_t addr)
```

Comment:
I really don't like this function. Many magic numbers, delicate pointer aritmetic. Can't it be done using struct fields?

Reply:
As suggested I changed the code to use struct fields (however I'll have to prepare a separate PR which moves 'hal' syspage part on top syspage struct).